src/network/mcpe/handler/InGamePacketHandler.php

```diff
@@ -869,6 +872,21 @@ public function handleLabTable(LabTablePacket $packet) : bool{
 		return false; //TODO
 	}
 
+	public function handleLecternUpdate(LecternUpdatePacket $packet) : bool{
+		//TODO: Cancel page turn event
+		//Drop book is handled on server side with an interact event.
+		//Total pages is gathered from book.
+
+		$lectern = $this->player->getWorld()->getBlockAt($packet->x, $packet->y, $packet->z);
```

Comment:
You should check that the target terrain is loaded, and also the distance (a player shouldn't be able to update a lectern a hundred blocks away).